Decode paths as VirtualPaths from -print-target-info output

owenv opened the pull request:
This change decodes the paths in `-print-target-info` as `VirtualPath`s to simplify their use throughout the driver. This initial change is pretty small, but I think it'll make it easier to use the runtime library and resource dir paths reported by the frontend throughout the rest of the driver. 

I thought about modifying the frontend so it always resolves these paths to absolute, but I think that would just create problems for distributed build systems in the future.

owenv commented:
@swift-ci test

DougGregor commented on Sources/SwiftDriver/Jobs/PrintTargetInfoJob.swift:
```diff
@@ -98,10 +98,30 @@ public struct FrontendTargetInfo: Codable {
+      case sdkPath, runtimeLibraryPaths, runtimeLibraryImportPaths, runtimeResourcePath
+    }
+
+    init(from decoder: Decoder) throws {
```
Somewhat sad to have to maintain these manually. Should we consider a VirtualPath wrapper of some sort that does Codable the way we want?

owenv replied:
Good idea, I added a `TextualVirtualPath` wrapper for this.

DougGregor reviewed (approved):
This does seem easier for clients of the target info, so +1 from me

owenv commented:
I made a couple changes in this version:
- `TextualVirtualPath` is a small wrapper to make decoding VirtualPaths from strings easier
- I added dynamic member lookup to avoid some unnecessary nesting when accessing members of FrontendTargetInfo.Paths
- rebased on the new file list changes

owenv commented:
@swift-ci test

owenv commented:
@swift-ci test

owenv commented:
@swift-ci please test

owenv commented:
Oops, I forgot about this one. Let's see if the tests still pass

@swift-ci test
